### PR TITLE
Added table for reorg blocks

### PR DESCRIPTION
**Pull request by DavidGershony**
Added storage of blocks that have been rewind and changed delete of unspent outputs to after the merge from inputs

### CODE REVIEW

**Review by sondreb — APPROVED** [review]
I'm not really able to review it properly, but the basics looks OK.

File: src/Blockcore.Indexer.Core/Storage/Mongo/MongoDb.cs [hunk 1]
```diff
@@ -126,5 +126,13 @@ public IMongoCollection<MempoolTable> Mempool
             return mongoDatabase.GetCollection<MempoolTable>("Mempool");
          }
       }
+
+      public IMongoCollection<ReorgBlockTable> ReorgBlock
```
Comment (dangershony) [inline src/Blockcore.Indexer.Core/Storage/Mongo/MongoDb.cs:130]: I think an alternative name is sometimes used OrphanBlock, what do you think?

File: src/Blockcore.Indexer.Core/Storage/Mongo/BlockRewindOperation.cs [hunk 2]
```diff
@@ -61,18 +65,44 @@ public static async Task RewindBlockAsync(this MongoData storage, long blockInde
       Task<DeleteResult> addressHistoryComputed =
          storage.AddressHistoryComputedTable.DeleteManyAsync(addrCompHistFilter);
 
+      await Task.WhenAll( output, transactions, addressComputed, addressHistoryComputed);
+
+      await MergeRewindInputsToUnspentTransactionsAsync(storage, blockIndex);
```
Comment (dangershony) [inline src/Blockcore.Indexer.Core/Storage/Mongo/BlockRewindOperation.cs:70]: So the order here is for the given block say block x that is reorged
- merge from `InputTable` all the deleted outputs by block X back to the `UnspentOutputTable`
- delete all the inputs of block X from the `InputTable`
- delete all the utxo added by block X from the `UnspentOutputTable`

File: src/Blockcore.Indexer.Core/Storage/Mongo/BlockRewindOperation.cs [hunk 3]
```diff
@@ -61,18 +65,44 @@ public static async Task RewindBlockAsync(this MongoData storage, long blockInde
...
+      var transactionIdsTask = storage.TransactionBlockTable.FindAsync(_ => _.BlockIndex == blockIndex);
 
-      FilterDefinition<InputTable> inputFilter =
-         Builders<InputTable>.Filter.Eq(addr => addr.BlockIndex, blockIndex);
+      Task.WhenAll(blockTask, inputsTask, outputsTask, transactionIdsTask);
 
-      await storage.InputTable.DeleteManyAsync(inputFilter);
+      var block = blockTask.Result.Single();
+
+      var reorgBlock = new ReorgBlockTable
```
Comment (dangershony) [inline src/Blockcore.Indexer.Core/Storage/Mongo/BlockRewindOperation.cs:95]: We should probably add another property here for the time the reorg happened.
Its a valid scenario that the same block will be reorged and go back to main chain and reorg again etc...
A network under attack with high malicious % of stake may experience that behavior

**Review by dangershony — APPROVED** [review]
This can go in but I recommend the time property as well but not urgent

**Pull request merged**